### PR TITLE
bug fixes for universal intrinsics of RISC-V back-end

### PR DESCRIPTION
**Pull request by joy2myself**
Fixed comparison operations and some packing operations.

All of the failures in core module for RISC-V with RVV.
See #20278 for previous state

Currently, all the core tests have passed:
```
[----------] Global test environment tear-down
[ SKIPSTAT ] 14 tests skipped
[ SKIPSTAT ] TAG='mem_6gb' skip 1 tests
[ SKIPSTAT ] TAG='skip_other' skip 13 tests
[==========] 11542 tests from 249 test cases ran. (3737861 ms total)
[  PASSED  ] 11542 tests.
```

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by joy2myself** [comment]
cc @asmorkalov

**Comment by asmorkalov** [comment]
Related: https://github.com/opencv/opencv/pull/20393

File: modules/core/include/opencv2/core/hal/intrin_rvv.hpp [hunk 1]
```diff
@@ -2321,28 +2323,23 @@ OPENCV_HAL_IMPL_RVV_SCAN_FORWOARD_OP(v_float64x2, double, f64)
...
-        ptr[3*i+1] = ptrvec[4*i+2]; \
-        ptr[3*i+2] = ptrvec[4*i+2]; \
-    } \
-    return v_load(ptr); \
+inline v_int8x16 v_pack_triplets(const v_int8x16& vec)
+{
+    uint64 CV_DECL_ALIGNED(32) ptr[2] = {0x0908060504020100, 0xFFFFFFFF0E0D0C0A};
+    return v_int8x16((vint8m1_t)vrgather_vv_u8m1((vuint8m1_t)vint8m1_t(vec), (vuint8m1_t)vle64_v_u64m1(ptr, 2), 16));
+}
+inline v_uint8x16 v_pack_triplets(const v_uint8x16& vec) { return v_reinterpret_as_u8(v_pack_triplets(v_reinterpret_as_s8(vec))); }
```
Comment (asmorkalov) [inline modules/core/include/opencv2/core/hal/intrin_rvv.hpp:2331]: Please move implementation to the new line as in function above.

File: modules/core/include/opencv2/core/hal/intrin_rvv.hpp [hunk 2]
```diff
@@ -1949,7 +1951,7 @@ inline void v_pack_store(_Tp* ptr, const _wTpvec& a) \
     _wTp CV_DECL_ALIGNED(32) arr[_Tpvec::nlanes] = {0}; \
```
Comment (asmorkalov) [inline modules/core/include/opencv2/core/hal/intrin_rvv.hpp:1951]: Memory usage looks very strange here. There are 32 vector registers and expect that universal intrisnic can be implemented without memory transactions.

File: modules/core/include/opencv2/core/hal/intrin_rvv.hpp [hunk 3]
```diff
@@ -1988,8 +1990,8 @@ inline void v_pack_u_store(_Tp* ptr, const _wTpvec& a) \
 { \
     _wTp CV_DECL_ALIGNED(32) arr[_Tpvec::nlanes] = {0}; \
```
Comment (asmorkalov) [inline modules/core/include/opencv2/core/hal/intrin_rvv.hpp:1991]: The same for memory here.

**Review by asmorkalov — CHANGES_REQUESTED** [review]

File: modules/core/include/opencv2/core/hal/intrin_rvv.hpp (hunk 1 above)
Comment (joy2myself) [reply modules/core/include/opencv2/core/hal/intrin_rvv.hpp]: resolved.

File: modules/core/include/opencv2/core/hal/intrin_rvv.hpp (hunk 2 above)
Comment (joy2myself) [reply modules/core/include/opencv2/core/hal/intrin_rvv.hpp]: I removed all the redundant CV_DECL_ALIGNED statements. They seem unnecessary. Now all the tests can still pass.

File: modules/core/include/opencv2/core/hal/intrin_rvv.hpp (hunk 3 above)
Comment (joy2myself) [reply modules/core/include/opencv2/core/hal/intrin_rvv.hpp]: The same as above.

File: modules/core/include/opencv2/core/hal/intrin_rvv.hpp (hunk 2 above)
Comment (alalek) [reply modules/core/include/opencv2/core/hal/intrin_rvv.hpp]: Just removal of `CV_DECL_ALIGNED(32)` modifier is a bad idea.
I believe you have asked to remove usage of memory buffer.

BTW, modern compilers are smart enough to eliminate unnecessary memory operations. Check generated assembly code of some test function to see result.

**Comment by asmorkalov** [comment]
@joy2myself Friendly reminder.

**Comment by joy2myself** [comment]
I have tried to optimize the memory usage in the whole implementation these days. The optimization also includes memory usage in the packaging operations mentioned above.  Code of  [this commit](https://github.com/joy2myself/opencv/commit/38ba74e565fdbb7591d58a7385b133f668165c7a) can be viewed as a reference for the current optimization.

However, most of these optimizations rely on the newly added native intrinsics `vset/vget` family. They have just been added by [the recent commit](https://github.com/riscv/rvv-intrinsic-doc/commit/485941bcdcc1c52324fd29088fb1287477c052fe) in rvv-intrinsic-doc. And they are not currently supported by the compiler (riscv-gnu-toolchian).

So my suggestion is, we merge the current implementation for now and optimize it later when rvv-intrinsics is updated to a new stable version and new native intrinsics are supported in the compiler. 

@asmorkalov  What do you think about it?

**Review by asmorkalov — APPROVED** [review]
:+1:  Let's go ahead with the current solution. Tested with QEMU, all tests are green. Let's do another optimization iteration, when set/get instructions are available in QEMU and compiler.

**Pull request merged**